py/kubeflow/testing/go-license-tools/patch_additional_license_info.py

```diff
@@ -0,0 +1,61 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+parser = argparse.ArgumentParser(
+    description='Generate dependencies json from license.csv file.')
+parser.add_argument(
+    'license_info_file',
+    nargs='?',
+    default='license_info.csv',
+    help=
+    'CSV file with license info fetched from github using '
+    'get-github-license-info CLI tool. (default: %(default)s)',
+)
+parser.add_argument(
+    'additional_license_info_file',
+    nargs='?',
+    default='additional_license_info.csv',
+    help=
+    'CSV file with license info. Each line is in the form '
+    '<license_url>,<license_type>. (default: %(default)s)',
+)
+args = parser.parse_args()
+
+
+def main():
+  mapping = {}
+  with open(args.additional_license_info_file, 'r') as f:
+    for line in f:
+      parts = line.strip().split(',')
+      assert len(parts) == 2
+      [license_url, license_type] = parts
+      mapping[license_url] = license_type
+
+  newlines = []
+  with open(args.license_info_file, 'r') as f:
+    for line in f:
+      parts = line.strip().split(',')
+      _, license_url, license_type, *_ = parts
+      if license_url in mapping and license_type == 'Other':
```

Comment:
For the purpose of the CI workflow, I think this script should better throw an error if some library has license_type 'Other', but it is not in the mapping.
This guards against incomplete patch file.

Reply:
Good suggestion! Done.